### PR TITLE
Add a tool for determining the licenses of dependencies.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

File: build/deplicense.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	continue
+    fi
+
+    toplevel=$(git -C "${dir}" rev-parse --show-toplevel 2>/dev/null)
+
+    git=1
+    if test "${toplevel}" = ""; then
+	toplevel=$(hg --cwd "${dir}" root 2>/dev/null)
+	if test "${toplevel}" = ""; then
+	    # TODO(pmattis): Handle subversion/bazaar.
```
Comment (bdarnell) [inline build/deplicense.sh]: You might be able to look at the glockfile for this; it always points to the repo root (although it also includes tools we use without linking into the binary so their licenses aren't as relevant).

File: build/deplicense.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	fi
+    done
+
+    echo "unable to determine license"
+}
+
+# List the current package and all of the dependencies which are not
+# part of the standard library (i.e. packages that contain a least one
+# dot in the first component of their name).
+pkgs=$(go list -f '{{printf "%s\n" .ImportPath}}{{range .Deps}}{{printf "%s\n" .}}{{end}}' . 2>/dev/null | \
```
Comment (bdarnell) [inline build/deplicense.sh]: Why is this different from `make listdeps`?

Comment (petermattis) [inline build/deplicense.sh]: This is copy&pasted from `depvers.sh`. I could probably use `make listdeps` here, but it seems a bit strange to make this script depend on the Makefile. I suppose I don't need the current package like `depvers.sh` does.

File: build/deplicense.sh (hunk 1 above)
Comment (petermattis) [inline build/deplicense.sh]: This is all taken from depvers.sh.

File: build/deplicense.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    return 1
+}
+
+function isApache2() {
+    grep -q '[[:space:]]*Apache License' "$1" && \
+	grep -q '[[:space:]]*Version 2\.0,' "$1"
+}
+
+function isBSD2Clause() {
+    grep -q 'Redistributions of source code' "$1" && \
```
Comment (bdarnell) [inline build/deplicense.sh]: This seems awfully broad; I wouldn't be surprised to see this text in other open-source licenses. 

Does anyone still use 4-clause BSD?

**Comment by bdarnell** [comment]
LGTM

File: build/deplicense.sh (hunk 3 above)
Comment (petermattis) [inline build/deplicense.sh]: Ok, I narrowed this down a bit and added detection for the BSD-4 clause license. None of our dependencies are using it, though.

File: build/deplicense.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    dirs+=("${toplevel}")
+    return 1
+}
+
+function isApache2() {
+    grep -q '[[:space:]]*Apache License' "$1" && \
+	grep -q '[[:space:]]*Version 2\.0,' "$1"
+}
+
+function isBSD2Clause() {
```
Comment (spencerkimball) [inline build/deplicense.sh]: There's no marker specifically mentioning BSD? or Berkeley or whatever?

Comment (petermattis) [inline build/deplicense.sh]: Nope. See http://opensource.org/licenses/BSD-3-Clause and http://opensource.org/licenses/BSD-2-Clause.

**Pull request merged**